Server/ConcordiaCurriculumManager/Program.cs

```diff
@@ -82,14 +84,18 @@ public static void Main(string[] args)
             .ReadFrom.Configuration(context.Configuration)
             .Enrich.FromLogContext();
         });
-        
+
         AddServices(builder.Services);
 
         builder.Services.AddMemoryCache();
         builder.Services.AddAutoMapper(typeof(Program));
 
-        builder.Services.AddControllers(options => {
+        builder.Services.AddControllers(options =>
+        {
             options.Filters.Add<ExceptionHandlerFilter>();
+        }).AddJsonOptions(options =>
+        {
+            options.JsonSerializerOptions.ReferenceHandler = ReferenceHandler.IgnoreCycles;

```

Comment:
This part is needed since a CourseCreationRequest references some courses and those courses also reference that CourseCreationRequest. Without this, the application crashes.